sui/manage-coins.js

```diff
@@ -0,0 +1,218 @@
+        if (CoinManager.isGasToken(coins.data[0])) {
+            coinObjectIds.shift();
+        }
+
+        if (coinObjectIds.length < 2) {
+            // Need at least 2 coins to merge
+            return;
+        }
+
+        const firstCoin = coinObjectIds.shift();
+        const remainingCoins = coinObjectIds.map((id) => tx.object(id));
+
+        tx.mergeCoins(firstCoin, remainingCoins);
+        console.log(`Merge ${coins.data.length} coins of type '${chalk.green(coins.data[0].coinType)}'`);
+    }
+
+    static async processCommand(config, chain, options) {
+        const [keypair, client] = getWallet(chain, options);
+        printInfo('Wallet Address', keypair.toSuiAddress());
+
+        const coinTypeToCoins = await CoinManager.getAllCoins(client, keypair.toSuiAddress());
+        CoinManager.printAllCoins(coinTypeToCoins);
+
+        const tx = new TransactionBlock();
+
+        if (options.merge) {
+            await CoinManager.mergeCoin(tx, coinTypeToCoins, options);
+        } else if (options.split) {
+            await CoinManager.splitCoins(tx, coinTypeToCoins, options);
+        }
+
+        const requireBroadcast = options.merge || options.split;
+
+        if (requireBroadcast) {
+            await client.signAndExecuteTransactionBlock({
+                transactionBlock: tx,
+                signer: keypair,
+                options: {
+                    showEffects: true,
+                    showObjectChanges: true,
+                    showContent: true,
+                },
+            });
+
+            console.log(`\nDone`);
+        }
+    }
+
+    static printAllCoins(coinTypeToCoins) {
+        console.log('==== Coins Info ====');
+
+        for (const coinType in coinTypeToCoins) {
+            const coins = coinTypeToCoins[coinType];
+            console.log(`Coin Type: ${chalk.green(coinType)}`);
+            console.log(`Total Balance: ${chalk.green(coins.totalBalance)}`);
+            console.log(`Total Objects: ${chalk.green(coins.data.length)}`);
+        }
+    }
+
+    static checkCoinType(coinType, coinTypeToCoins) {
+        if (coinType && !coinTypeToCoins[coinType]) {
+            console.error(`No coins found for coin type ${coinType}`);
+            process.exit(0);
+        }
+    }
+
+    static isGasToken(coin) {
+        return coin.coinType === '0x2::sui::SUI';
+    }
+
+    static checkSplitAmount(options) {
+        try {
+            parseInt(options.split);
+        } catch (e) {
+            console.error('\nError: Please specify a valid split amount');
+            process.exit(0);
+        }
+    }
+}
+
+async function mainProcessor(options, processor) {
+    const config = loadSuiConfig(options.env);
+    await processor(config, config.sui, options);
+    saveConfig(config, options.env);
+}
+
+if (require.main === module) {
+    const program = new Command();
+    program.name('manage-coins').description('Merge or split coins for an account');
```

Comment:
```suggestion
    const program = new Command();

    program.name('tokens').description('Token management tool (e.g. show balance, merge, split etc.)');
```
Let's standardize `tokens.js` for all chain folders?